### PR TITLE
two small issues in cursor and sharing

### PR DESCRIPTION
**Pull request by velikaww**
1. if cursor got disabled and enabled later at another surface, state may not appear correctly as the common sense. this is because target get set to null when disabled but not recover/update at enable.
2. in debug mode, create local anchor process often stuck without located or any event triggered. changed back to origin code version, letting the local anchor created on demand instead of always during Start phase.

### CODE REVIEW

File: Assets/HoloToolkit/Sharing/Tests/Scripts/ImportExportAnchorManager.cs [hunk 1]
```diff
@@ -701,13 +693,14 @@ private void AnchorStoreReady(WorldAnchorStore store)
         /// </summary>
         private void CreateAnchorLocally()
         {
-            if (thisAnchor.isLocated)
+            WorldAnchor anchor = this.EnsureComponent<WorldAnchor>();
```
Comment (thebanjomatic) [inline Assets/HoloToolkit/Sharing/Tests/Scripts/ImportExportAnchorManager.cs]: I don't think we need to be calling `EnsureComponent` (and by extension `GetComponent`) every frame until `anchor.isLocated` returns `true`. I'd like to see a sort of hybrid between the current code and the version you have here:  

```cs
if (thisAnchor == null)
{
    thisAnchor = this.EnsureComponent<WorldAnchor>();
}
if (thisAnchor.isLocated)
{
    ...
}
```

That way we are still initializing the `WorldAnchor` component on-demand, but we only wind up calling `EnsureComponent` once.

File: Assets/HoloToolkit/Sharing/Tests/Scripts/ImportExportAnchorManager.cs [hunk 2]
```diff
@@ -883,11 +876,12 @@ private void AnchorLoadComplete()
         /// </summary>
         private void Export()
         {
+            WorldAnchor anchor = this.GetComponent<WorldAnchor>();
```
Comment (thebanjomatic) [inline Assets/HoloToolkit/Sharing/Tests/Scripts/ImportExportAnchorManager.cs]: In accordance with the suggested change above, the changes in this function should be able to be reverted.

File: Assets/HoloToolkit/Sharing/Tests/Scripts/ImportExportAnchorManager.cs (hunk 1 above)
Comment (StephenHodgson) [inline Assets/HoloToolkit/Sharing/Tests/Scripts/ImportExportAnchorManager.cs]: Agreed

Actually, I think that was the original reason why I moved `thisAnchor` into an outer scope.

Comment (yacuzo) [inline Assets/HoloToolkit/Sharing/Tests/Scripts/ImportExportAnchorManager.cs]: I think I had some problem with the anchor only being crated once, in the following situation:
* Anchor is crated
* Anchor is located
* Anchor is serialized, but is rejected due to small size
* Anchor is never re-crated, so the serialization size is always the same.

If the anchor is re-crated the definition is also re-crated, and if the spatial map is now bigger the serialization will also be more accurate (and be a larger data blob). I'm not sure if re-serializing is enough when the spatial map is improved, but i got the feeling a re-creation of the anchor was necessary.

Comment (yacuzo) [inline Assets/HoloToolkit/Sharing/Tests/Scripts/ImportExportAnchorManager.cs]: I do agree that Get/AddComponent should not be called every frame, it is slow. It should probably be done once, maybe in Connected(), and be re-crated wherever a failed serialization is handled.

**Comment by StephenHodgson** [comment]
I actually would like to suggest breaking this PR into two requests. One for the cursor and the other for the anchor.

**Comment by velikaww** [comment]
Sorry to be quite busy last few days and having not token a look here.
OK I will split this PR to be two,  For the latter anchor one, i actually simply revert to what it last working status, and also I don't thing the cost GetComponet is significant enough that time. Further, what @yacuzo mentioned should better be considered, we'd better re-create the anchor in this case.

**Comment by velikaww** [comment]
hmm... Seems I cannot create another PR, Is that any limitation? I am not quite familiar with GitHub contribution process.

**Comment by StephenHodgson** [comment]
No limitations on the number of PRs you can submit.

**Comment by velikaww** [comment]
Let me try close this one first and then raise another...

**Comment by velikaww** [comment]
OK. then I am able to create another PR to this project, seems there could be only one PR between two repoes (maybe regards containing same commit). Please review #604.
And for the Anchor one, I will make it tomorrow or the day after.

**Comment by StephenHodgson** [comment]
Correct, the changes need to be on separate branches if you wish to submit at the same time.

**Comment by velikaww** [comment]
BTW, I just read the code again, after more than 10 days from when I made those changes. The method CreateAnchorLocally will not be called every frame, so I don't think we should care the EnsureComponent cost of a "once-per-initial-or-fail" action (Including getComponemt in Export). Here, seems not necessary to elevate the anchor variable to be class-wise.
What we should consider to improve is @yacuzo mentioned, whether remove the component on fail.

**Pull request closed without merging**